chain-patrol-ts/src/agent.ts

```diff
@@ -0,0 +1,99 @@
+import { BlockEvent, Finding, Initialize, HandleBlock } from "forta-agent";
+import { fetchApiInfo, getCurrentDateInYyyyMmDD } from "./utils";
+import { ApiInfo, Asset, AssetDetails, UnalertedAsset } from "./types";
+import { createBlockedAssetFinding } from "./findings";
+import AssetFetcher from "./fetcher";
+import {
+  ETHEREUM_BLOCKS_IN_ONE_DAY,
+  MAX_ASSET_ALERTS_PER_BLOCK,
+  ASSET_BLOCKED_STATUS,
+  INIT_API_QUERY_DATE,
+  ASSET_DETAILS_URL,
+  ASSET_LIST_URL,
+} from "./constants";
+
+let assetFetcher: AssetFetcher;
+let apiQueryStartDate: string;
+
+// Bots are allocated 1GB of memory, so storing
+// `UnalertedAsset`s won't be an issue. Especially
+// since entries will be cleared after alerted.
+const unalertedAssets: UnalertedAsset[] = [];
+
+async function createAssetFetcher(
```

Comment:
defining this function here and giving it as an argument to `provideInitialize` seemed awkward where you could just define this function inside the initialize block. if it's for mocking purposes, right now I'm not sure how to mock it anyway. I'll think about it.

Reply:
Agreed that it is a bit _awkward_, but as you bring up, it is a current workaround for mocking. We implemented something similar for the Solidus bot [here](https://github.com/forta-network/forta-solidus-bot/blob/main/src/agent.ts#L16-L19).